### ISSUE TITLE
Is 8.3.0 has changes in abi?

### ISSUE THREAD

**Issue by Vascom** [weidai11/cryptopp#987]
I am maintainer of Crypto++ package in Fedora. I see that programs built with 8.2.0 failed to run with 8.3.0 without rebuild but soname version not changed.

So I need to know if abi changed and I need to rebuild all depended packages?

Example of the error:
```
megasync: symbol lookup error: megasync: undefined symbol: _ZNK8CryptoPP21KeyDerivationFunction16MinDerivedLengthEv
```

**Comment by Vascom** [comment]
Thanks.

I will rebuild all depended packages. It is seems not a problem. Just for understanding.

**Comment by noloader** [comment]
@Vascom,

How did you make out? Do we need to keep this issue open?

**Comment by Vascom** [comment]
Issue can be closed.

**Comment by noloader** [comment]
@Vascom

Ack, thanks. Keep those email addresses handy. I'm happy to help over email if that is more convenient for you.

**Comment by thesamesam** [comment]
Thanks to @noloader for finding a downstream [bug](https://bugs.gentoo.org/762241) before we'd even worked out the problem, it's really appreciated.

It looks like there may indeed be some ABI [breakage](https://abi-laboratory.pro/index.php?view=compat_report&l=cryptopp&v1=8_2_0&v2=8_3_0&obj=e8fc4&kind=abi#Type_Problems_High) in 8.3.0. A user reported amule was broken until it was recompiled against 8.3.0. The linked report mentions some changes: both dropped symbols but also changes in structure size which can manifest as (seemingly random) runtime breakage.

**Comment by thesamesam** [comment]
@noloader @mouse07410 Would it be possible to reopen the issue for now to ensure it's not forgotten?

**Comment by noloader** [comment]
I think we can close this now. Crypto++ 8.5 was released, and Crypto++ 8.6 is comping up soon.
